full-moon/src/atom.rs

```diff
@@ -0,0 +1,295 @@
+use logos::{Lexer, Logos};
+
+use crate::ShortString;
+
+pub fn trim_bracket_head(slice: &str) -> (ShortString, Option<usize>) {
```

Comment:
Visibility: is this necessary?
```suggestion
pub(crate) fn trim_bracket_head(slice: &str) -> (ShortString, Option<usize>) {
```